lib/Controller/DocumentRevisionController.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * @author Piotr Mrowczynski <piotr@owncloud.com>
+ *
+ * @copyright Copyright (c) 2023, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OCA\Richdocuments\Controller;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\Files\InvalidPathException;
+use OCP\Files\IRootFolder;
+use OCP\Files\Storage\IVersionedStorage;
+use OCP\IRequest;
+use OCP\IUserSession;
+
+class DocumentRevisionController extends Controller {
+	/**
+	 * @var IUserSession The user session service
+	 */
+	private $userSession;
+
+	/**
+	 * @var IRootFolder The root folder service
+	 */
+	private $rootFolder;
+
+	public function __construct(
+		string $appName,
+		IRequest $request,
+		IUserSession $userSession,
+		IRootFolder $rootFolder
+	) {
+		parent::__construct($appName, $request);
+		$this->userSession = $userSession;
+		$this->rootFolder = $rootFolder;
+	}
+
+	/**
+	 * Get collabora document revisions for:
```

Comment:
Not sure if it's worthy to say that the current (latest) version of the file will be excluded (or so it seems).

Reply:
done